pkg/controller/mpi_job_controller.go

```diff
@@ -985,14 +1001,15 @@ func (c *MPIJobController) updateMPIJobStatus(mpiJob *kubeflow.MPIJob, launcher
 			mpiJob.Status.StartTime = &now
 		}
 	}
-	launcherPods, err := c.jobPods(launcher)
-	if err != nil {
-		return fmt.Errorf("checking launcher pods running: %w", err)
-	}
-	// Job.status.Active accounts for Pending and Running pods. Count running pods
-	// from the lister instead.
-	launcherPodsCnt := countRunningPods(launcherPods)
+        launcherPodsCnt := 0
 	if launcher != nil {
+		launcherPods, err := c.jobPods(launcher)
+		if err != nil {
+			return fmt.Errorf("checking launcher pods running: %w", err)
+		}
+		// Job.status.Active accounts for Pending and Running pods. Count running pods
+		// from the lister instead.
+		launcherPodsCnt := countRunningPods(launcherPods)
```

Comment:
```suggestion
		launcherPodsCnt = countRunningPods(launcherPods)
```

This seems to have caused a test failure.

Reply:
@xhejtman